src/planscape/collaboration/views.py

```diff
@@ -46,3 +50,20 @@ def post(self, request, format=None):
             logger.exception("Something failed during the creation of a new invite.")
             # this will return a 500
             raise
+
+
+class UserObjectRolesForObject(APIView):
+
+    def get(self, request: Request, target_entity: str, object_pk: int):
+
+        user = request.user
+        if not check_for_permission(user.pk, "add_collaborator"):
```

Comment:
I think here you should probably use `CollaboratorPermission.can_add` instead, as you are not including the case of the user being the creator